typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -317,7 +327,45 @@ export class EvmERC20WarpRouteReader extends EvmRouterReader {
               error,
             );
           }
+
+          try {
+            const maybeEverclearTokenBridge =
+              EverclearTokenBridge__factory.connect(
+                warpRouteAddress,
+                this.provider,
+              );
+
+            await maybeEverclearTokenBridge.callStatic.everclearAdapter();
+
+            let everclearTokenType = TokenType.collateralEverclear;
+            try {
```

Comment:
personally not a huge fan of these nested try/catch blocks as it becomes hard to read but I recognize that there is no obvious alternative

Reply:
yeah hopefully we can revisit when we move the EVM code to the multi vm format